bootstrap-cluster.sh

```diff
@@ -424,19 +454,34 @@ if [ "$K8S_SNODE" == "true" ]; then
     :  # Do nothing
 
 else
+    scp -i "$KEY" -o StrictHostKeyChecking=no \
+        -o ProxyCommand="ssh -o StrictHostKeyChecking=no -i $KEY -W %h:%p root@${BASTION_IP}" \
+        "$KEY" root@${mnodes[0]}:/tmp/tmpkey.pem
+
     ssh -i "$KEY" -o StrictHostKeyChecking=no \
         -o ProxyCommand="ssh -o StrictHostKeyChecking=no -i \"$KEY\" -W %h:%p root@${BASTION_IP}" \
         root@${mnodes[0]} "
     MANGEMENT_NODE_IP=${mnodes[0]}
+    chmod 400 /tmp/tmpkey.pem
     for node in ${storage_private_ips}; do
+        echo \"\"
+        echo \"Getting PCIe address on node \${node} \"
+        echo \"\"
+
+        PCIE=\$(ssh -i /tmp/tmpkey.pem -o StrictHostKeyChecking=no root@\$node \"lspci -D | grep -i 'NVM' | awk '{print \\\$1}' | paste -sd ' ' -\")
+
+        echo \"PCIe: \$PCIE\"
+
         echo ""
         echo "joining node \${node}"
-        add_node_command=\"${command} ${CLUSTER_ID} \${node}:5000 eth0 --data-nics eth1\"
+        add_node_command=\"${command} ${CLUSTER_ID} \${node}:5000 eth0 --data-nics eth1 --ssd-pcie \$PCIE\"
         echo "add node command: \${add_node_command}"
         \$add_node_command
         sleep 3
     done
 
+    rm -f /tmp/tmpkey.pem
```

Comment:
Create a cleanup function at the top and use TRAP exit to cleanup the key

cleanup() {
  echo "Cleaning up temp key..."
  rm -f "$TEMP_KEY"
}
trap cleanup EXIT INT TERM

Reply:
I have made the suggested fix